File: base/fizz-buzz/fizz-buzz.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

import unittest

def fizz_buzz(n):
    output = []
    for i in range(1, n+1):
        if i % 15 == 0:
            output += ['Fizz Buzz']
        elif i % 3 == 0:
            output += ['Fizz']
        elif i % 5 == 0:
            output += ['Buzz']
        else:
            output += [i]
    return output

class TestMerge(unittest.TestCase):
    def test_solve(self):
        self.assertEqual(fizz_buzz(15), [1, 2, 'Fizz', 4, 'Buzz', 'Fizz', 7 , 8 ,'Fizz' , 'Buzz', 11, 'Fizz' ,13 , 14, 'Fizz Buzz'])
if __name__ == '__main__':
    unittest.main()

File: base/fizz-buzz/fizz-buzz.rb
def fizz_buzz(num)
    if num % 15 == 0
        p num.to_s + ": Fizz Buzz"
    elsif num % 3 == 0
        p num.to_s +  ": Fizz"
    elsif num % 5 == 0
        p num.to_s + ": Buzz" 
    end
end


(1..100).map {|num| fizz_buzz(num) }
